docs/docs/building.md

```diff
@@ -4,158 +4,208 @@ title: Building an upgradeable application
+
+    zos create Basil --network development --init --args $OWNER
+
+Let's fix our bug. Edit `contracts/Basil.sol` to add the missing line to the `donate` function:
+
+    function donate(uint256 _red, uint256 _green, uint256 _blue) public payable {
+      require(_red < 256);
+      require(_green < 256);
+      require(_blue < 256);
+      require(msg.value > highestDonation);
+      red = _red;
+      green = _green;
+      blue = _blue;
+      highestDonation = msg.value;
+      emit NewDonation(msg.sender, msg.value, red, green, blue);
+    }
+
+And to finish our fix, we compile the patched contract, sync with ZeppelinOS and upgrade the proxy:
+
+    zos upgrade Basil <proxy_address_1> --network development
+
+TODO: where does the proxy address comes from?
+
+## Upgrading the Migratable initialize
+
+Another common thing that happens when developing smart contracts for Ethereum is that new standards appear, all the new kids implement them in their contracts, and a very cool synergy between contracts starts to happen. The people who have immutable contracts already deployed will miss all the fun. This has just happened to us: it would be very nice to encourage donations to Basil by emitting a unique ERC721 token in exchange. Well, let's upgrade the contract with ZeppelinOS to do just that.
+
+We could modify `contracts/Basil.sol` as before. But now let's try something different. Let's make a new contract in `contracts/BasilERC721.sol`, that inherits from our initial version of Basil:
+
+    pragma solidity ^0.4.23;
+
+    import "./Basil.sol";
+    import "openzeppelin-zos/contracts/token/ERC721/MintableERC721Token.sol";
+
+    contract BasilERC721 is Basil {
+      using SafeMath for uint256;
+
+      // ERC721 non-fungible tokens to be emitted on donations.
+      MintableERC721Token public token;
+      uint256 public numEmittedTokens;
+
+      function initialize(MintableERC721Token _token) public isInitializer("Basil", "1") {
+        require(_token != address(0));
+        require(token == address(0));
+        token = _token;
+      }
+
+      function donate(uint256 _red, uint256 _green, uint256 _blue) public payable {
+        super.donate(_red, _green, _blue);
+        emitUniqueToken(tx.origin);
+      }
+
+      function emitUniqueToken(address _tokenOwner) internal {
+        token.mint(_tokenOwner, numEmittedTokens);
+        numEmittedTokens = numEmittedTokens.add(1);
+      }
+    }
+
+A few things to note:
+ * This new version extends from the previous one. This is a very handy pattern, because the proxy used in ZeppelinOS requires new versions to preserve the state variables. TODO why? link to the FAQ maybe?
```

Comment:
Remove TODO. We'll see where we include this later.

Reply:
made a question on stackexchange for this. If it later turns into a frequent question, we can add the link.